### DISCUSSION TITLE
Using command pallet within `tasks.json`

### DISCUSSION THREAD

**Discussion by jtlandis** [zed-industries/zed#16775] [Feature Requests]
Is there a way to use one of the command pallet options from within a tasks.json task?

For the scope, I would like to construct a `runnables.scm` file for markdown/rmd/qmd files based on [this zed decoded post](https://zed.dev/blog/zed-decoded-tasks).

What I'm imagining is that the runnables.scm query would highlight all fenced code content (somehow?) and then execute repl::run

```scm
// within runnables.scm
(
  ((fenced_code_block) @run
    (fenced_code_block_delimiter)
    (info_string
      (language) 
    ((code_fence_content) @repl_content)
    (fenced_code_block_delimiter)
    )
) @run-repl-chunk
```

```json
// within tasks.json
{
  "label": "run repl chunk",
  "command": "repl:run $ZED_CUSTOM_repl_content",
  "tags": ["run-repl-chunk"]
}

```

To make the above completely work, I imagine there must also be a way to set current selection via \\$ZED_CUSTOM_\\$(capture_name) (based on [languages extension](https://zed.dev/docs/extensions/languages#runnable-code-detection)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
